tests/vmcheck/test-basic.sh

```diff
@@ -59,3 +59,17 @@ echo "ok status doesn't require root"
 # Also check that we can do status as non-root non-active
 vm_cmd runuser -u bin rpm-ostree status
 echo "ok status doesn't require active PAM session"
+
+# Add metadata string containing EnfOfLife attribtue
+META_ENDOFLIFE_MESSAGE="this_is_a_test"
+commit=$(vm_cmd ostree commit -b vmcheck \
+            --tree=ref=vmcheck --add-metadata-string=ostree.endoflife=$META_ENDOFLIFE_MESSAGE)
```

Comment:
the above implementation seems not functioning,( which I have no clue why), it seems like when I have the above implementation, there is a random single quote get introduced.
Here is the test output with the above implementation, 
https://paste.fedoraproject.org/paste/fNGo0KGnJKH~2Q3VSRvOTQ

However, the implementation  below seems to pass the test, 
**`--add-metadata-string=ostree.endoflife=\"${META_ENDOFLIFE_MESSAGE}\"`**,
but seeing the testing output, it makes me more confused
https://paste.fedoraproject.org/paste/OAenqkrjH-MALc8m6MPzhw, some single quote gets inserted again. 

I was trying to look for some problemssimilar online, but was unable to do so, is there a possible reason for this behaviour(introduction of single quotes) to happen?